test/functional/address_types.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that the wallet can send and receive using all combinations of address types."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class AddressTypeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
+        self.setup_clean_chain = True
+        self.extra_args = [["-addresstype=legacy"], ["-addresstype=p2sh"], ["-addresstype=bech32"], ["-addresstype=p2sh", "-changetype=bech32"]]
+
+    def run_test(self):
+        self.nodes[0].generate(120)
```

Comment:
Doesn't the test framework already give you plenty of mature coins out of the box? cc @jnewbery

Reply:
@Sjors setup_clean_chain was set to True in this test https://github.com/bitcoin/bitcoin/pull/11403/files/e2e9ead25fe0ddb364e35f8eabb5a7f937d3973b#diff-e3da38b280555e81d6b794e1117ed0a8R12

Reply:
I see. But why? Doesn't that slow down the tests?